src/components/Inputs/AmplifyComboBox/AmplifyComboBoxMenuItem.tsx

```diff
@@ -0,0 +1,244 @@
+import { KeyboardEvent, MouseEvent, useMemo, useRef, useState } from 'react';
+
+import { Icon, Menu } from '@equinor/eds-core-react';
+import {
+  arrow_drop_down,
+  arrow_drop_up,
+  checkbox,
+  checkbox_indeterminate,
+  checkbox_outline,
+} from '@equinor/eds-icons';
+import { tokens } from '@equinor/eds-tokens';
+
+import {
+  MenuItemMultiselect,
+  MenuItemParentSelect,
+  MenuItemSpacer,
+} from './AmplifyComboBox.styles';
+import {
+  AmplifyComboBoxMenuProps,
+  ComboBoxOption,
+} from './AmplifyComboBox.types';
+import { getChildOffset } from './AmplifyComboBox.utils';
+
+const { colors } = tokens;
+
+interface AmplifyComboBoxMenuItemProps<T> {
+  item: T;
+  index: number;
+  childOffset: number;
+  depth?: number;
+}
+
+type AmplifyComboBoxSingleSelectMenuItemProps<T extends ComboBoxOption<T>> = {
```

Comment:
Should these types also be moved to the .types file? Since the Menu types are there

Reply:
Pushed a commit now